SPECS/curl/curl.spec

```diff
@@ -1,15 +1,13 @@
 Summary:        An URL retrieval utility and library
 Name:           curl
-Version:        8.3.0
-Release:        2%{?dist}
+Version:        8.5.0
```

Comment:
Lets try to build the dependents as well before we merge.

Last I recall, there used to be a few `perl-*` packages that complained whenever we did a big jump to curl.

Reply:
To be clear, you mean dash-rolling things that BuildRequires any curl package/subpackage? Should I perform this step for the (already merged) 2.0 PR as well? https://github.com/microsoft/CBL-Mariner/pull/7059

Reply:
No sir .. not a dash-roll, just a sanity check to see if everything builds as it should :)
It most likely would, but just a double check :)

Reply:
I have crafted this spell to get a list of packages to buddybuild.

`grep -Rl 'BuildRequires:.*curl' SPECS/ | xargs -I % bash -c 'basename $(dirname "%")' | tr '\n' ' '`

See the following buddybuild: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=475045&view=results